doc/reference/power_management/index.rst

```diff
@@ -403,6 +403,45 @@ Check Busy Status of All Devices API
 
 Checks if any device is busy. The API returns 0 if no device in the system is busy.
 
+Wakeup capability
+-----------------
+
+Some devices are capable of wake the system up from a sleep state.
+Such devices have the bit :c:macro:`PM_DEVICE_ATOMIC_FLAGS_WS_CAPABLE_BIT` set
+in :c:member:`pm_device.atomic_flags`. When a device is "wakeup" capable, applications
+can enable or disable this feature on a device dynamically using
+:c:func:`pm_device_wakeup_enable`.
+
+This property can be set on device declaring the property ``wakeup-source`` in
+the device node in devicetree.  For example, this devicetree fragment sets
+``gpiol@40022000`` as "wakeup" capable:
+
+.. code-block:: devicetree
+
+		gpio0: gpio@40022000 {
+			compatible = "ti,cc13xx-cc26xx-gpio";
+			reg = <0x40022000 0x400>;
+			interrupts = <0 0>;
+			status = "disabled";
+			label = "GPIO_0";
+			gpio-controller;
+			wakeup-source;
+			#gpio-cells = <2>;
+		};
+
```

Comment:
Considering that wakeup-sources are enabled by default, I don't think it is a good idea to place the flag in soc dts files, so I think:

- They need to be manually enabled, or
- The application sets the flag if needed, that is:
```
&gpio0 {
    wakeup-source;
};
```

Reply:
On my side, I'd say this is a good thing to be able to identify devices that are "wakeup source" capable in device tree.
Then, I'd suggest to disable them by default: treat them as other devices unless required by application.